### PR TITLE
fix(datepicker): update input size and invalid class on calendar click

### PR DESCRIPTION
**Pull request by mhchen**
@jelbourn Please review

### CODE REVIEW

**Comment by mhchen** [comment]
Also updated a few unit tests involving the invalid class to use the .toHaveClass Jasmine matcher that I should've seen the first time around.

File: src/components/datepicker/datePicker.js [hunk 1]
```diff
@@ -212,8 +212,11 @@
 
     self.$scope.$on('md-calendar-change', function(event, date) {
       self.ngModelCtrl.$setViewValue(date);
+      self.date = date;
       self.inputElement.value = self.dateLocale.formatDate(date);
       self.closeCalendarPane();
+      self.resizeInputElement();
+      self.inputContainer.classList.remove(INVALID_CLASS);
```
Comment (jelbourn) [inline src/components/datepicker/datePicker.js]: Would it work if you just called ngModelCtrl.$render here?

Comment (mhchen) [inline src/components/datepicker/datePicker.js]: Hm no, it doesn't look like calling $render actually triggers the input handlers

**Comment by jelbourn** [comment]
LGTM

**Pull request closed without merging**